app/src/main/java/com/github/onlynotesswent/ui/overview/EditMarkdownScreen.kt

```diff
@@ -0,0 +1,303 @@
+ * markdown file is created and uploaded, then re-downloaded.
+ *
+ * @param navigationActions Object to handle navigation actions, including going back to the
+ *   previous screen.
+ * @param noteViewModel ViewModel to manage the note's state and interactions.
+ * @param userViewModel ViewModel for user-specific actions, if required.
+ * @param fileViewModel ViewModel to handle file downloads and uploads for markdown files.
+ */
+@Composable
+fun EditMarkdownScreen(
+    navigationActions: NavigationActions,
+    noteViewModel: NoteViewModel,
+    userViewModel: UserViewModel,
+    fileViewModel: FileViewModel
+) {
+  val state = rememberRichTextState()
+
+  val context = LocalContext.current
+  val note by noteViewModel.selectedNote.collectAsState()
+  var markdownContent: File? by remember { mutableStateOf(null) }
+
+  // Function to download and set the Markdown file
+  LaunchedEffect(Unit) {
+    fileViewModel.downloadFile(
+        uid = note?.id ?: "errorNoId",
+        fileType = FileType.NOTE_TEXT,
+        context = context,
+        onSuccess = { downloadedFile: File ->
+          // Update the UI with the downloaded file reference
+          markdownContent = downloadedFile
+          state.setMarkdown(markdownContent?.readText() ?: "")
+        },
+        onFailure = { exception ->
+          Toast.makeText(context, "Error downloading file: ${exception.message}", Toast.LENGTH_LONG)
+              .show()
+        })
+  }
+  @Suppress("kotlin:S6300") // as there is no need to encrypt file
+  // Function to update the Markdown file with "Hello world"
+  fun updateMarkdownFile(context: Context, uid: String, fileViewModel: FileViewModel) {
+    try {
+      if (markdownContent != null) {
+        // Write "Hello world" to the file
+
+        markdownContent!!.writeText(state.toMarkdown())
+
+        // Get the file URI
+        val fileUri = Uri.fromFile(markdownContent)
+
+        // Upload the file to update it in Firebase Storage
+        fileViewModel.updateFile(uid = uid, fileUri = fileUri, fileType = FileType.NOTE_TEXT)
+      }
+    } catch (e: IOException) {
+      Toast.makeText(context, "Error updating file: ${e.message}", Toast.LENGTH_LONG).show()
+    }
+  }
+
+  Scaffold(
+      topBar = {
+        ScreenTopBar(
+            titleTestTag = "modifyMDTitle",
+            onBackClick = { navigationActions.goBack() },
+            title = "Modify Markdown",
+            icon = {
+              Icon(imageVector = Icons.AutoMirrored.Outlined.ArrowBack, contentDescription = "Back")
+            },
+            iconTestTag = "goBackButton")
+      },
+      content = { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(16.dp)
+                    .padding(paddingValues)
+                    .testTag("editMarkdownColumn")
+                    .verticalScroll(rememberScrollState()),
```

Comment:
Since the contents of the column are limited, `Modifier.verticalScroll()` should be fine but maybe consider the use of a `LazyColumn` as it is more optimized for scrolling tasks. The changes are minimal, change column to lazy column and call the column members inside` item{ }` objects